mmtk/src/active_plan.rs

```diff
@@ -5,7 +5,34 @@ use mmtk::util::opaque_pointer::*;
 use mmtk::vm::ActivePlan;
 use mmtk::Mutator;
 use mmtk::Plan;
-use std::sync::Mutex;
+use std::sync::{Mutex, MutexGuard};
+
+struct OpenJDKMutatorIterator<'a> {
+    _guard: MutexGuard<'a, ()>,
+}
+
+impl<'a> OpenJDKMutatorIterator<'a> {
+    fn new(guard: MutexGuard<'a, ()>) -> Self {
+        // Reset mutator iterator
+        unsafe {
+            ((*UPCALLS).reset_mutator_iterator)();
+        }
+        Self { _guard: guard }
+    }
+}
+
+impl<'a> Iterator for OpenJDKMutatorIterator<'a> {
+    type Item = &'a mut Mutator<OpenJDK>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let next = unsafe { ((*UPCALLS).get_next_mutator)() };
```

Comment:
This line is the same.  If we encapsulate the `JavaThreadIteratorWithHandle` instance in the `OpenJDKMutatorIterator` itself, we don't need to access the global iterator instance from here.